components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -179,7 +206,12 @@ fn in_union(implicit_args: &[ScalarValue]) -> bool {
     implicit_args.get(0) == Some(&ScalarValue::Int(Some(1)))
 }
 
-/// The unsigned int implementation for push down signature `CastIntAsDecimal`.
+fn is_field_type_unsigned(ft: &FieldType) -> bool {
+    ft.flag().contains(FieldTypeFlag::UNSIGNED)
+}
+
+// -------------------* as decimal----------------------------------
+// TODO, TiDB's impl has bug, fix this after fixed TiDB's
```

Comment:
I think we should not add this comment about TiDB has BUG (it's better to file an issue in TiDB repo).